<div align="center">

 <img src="https://upload.wikimedia.org/wikipedia/fr/thumb/8/80/Universit%C3%A9_Panth%C3%A9on-Assas_%28depuis_2016%29.svg/1280px-Universit%C3%A9_Panth%C3%A9on-Assas_%28depuis_2016%29.svg.png" width="50%" height="50%"> 
  <img src="https://cdn.pixabay.com/photo/2017/09/08/19/05/a-2729781_1280.png" width="50%" height="50%"> 


<h1>TD 1 de Machine learning</h1>


 </div>





# 1 Importation, description et visualisation des données

## 1 Importez les bases de données en utilisant le package pandas

In [1]:
import pandas
X = pandas.read_excel("https://github.com/gregory798/ML_MBFA/blob/main/TD1/data_X.xlsx?raw=true")
y = pandas.read_excel("https://github.com/gregory798/ML_MBFA/blob/main/TD1/data_y.xlsx?raw=true")

## 2 Décrivez la structure des données

In [12]:
X

pixel1  pixel2  pixel3  pixel4  pixel5  pixel6  pixel7  pixel8  pixel9  \
0          0       0       0       0       0       0       0       0       0   
1          0       0       0       0       0       0       0       0       0   
2          0       0       0       0       0       0       0       0       0   
3          0       0       0       0       0       0       0       0       0   
4          0       0       0       0       0       0       0       0       0   
...      ...     ...     ...     ...     ...     ...     ...     ...     ...   
4995       0       0       0       0       0       0       0       0       0   
4996       0       0       0       0       0       0       0       0       0   
4997       0       0       0       0       0       0       0       0       0   
4998       0       0       0       0       0       0       0       0       0   
4999       0       0       0       0       0       0       0       0       0   

      pixel10  ...  pixel775  pixel776  pixel777  pixel778  pixel779  \
0           0  ...         0         0         0         0         0   
1           0  ...         0         0         0         0         0   
2           0  ...         0         0         0         0         0   
3           0  ...         0         0         0         0         0   
4           0  ...         0         0         0         0         0   
...       ...  ...       ...       ...       ...       ...       ...   
4995        0  ...         0         0         0         0         0   
4996        0  ...         0         0         0         0         0   
4997        0  ...         0         0         0         0         0   
4998        0  ...         0         0         0         0         0   
4999        0  ...         0         0         0         0         0   

      pixel780  pixel781  pixel782  pixel783  pixel784  
0            0         0         0         0         0  
1            0         0         0         0         0  
2            0         0         0         0         0  
3            0         0         0         0         0  
4            0         0         0         0         0  
...        ...       ...       ...       ...       ...  
4995         0         0         0         0         0  
4996         0         0         0         0         0  
4997         0         0         0         0         0  
4998         0         0         0         0         0  
4999         0         0         0         0         0  

[5000 rows x 784 columns]

In [5]:
print(type(X))

<class 'pandas.core.frame.DataFrame'>


In [6]:
print(X.head(10))

   pixel1  pixel2  pixel3  pixel4  pixel5  pixel6  pixel7  pixel8  pixel9  \
0       0       0       0       0       0       0       0       0       0   
1       0       0       0       0       0       0       0       0       0   
2       0       0       0       0       0       0       0       0       0   
3       0       0       0       0       0       0       0       0       0   
4       0       0       0       0       0       0       0       0       0   
5       0       0       0       0       0       0       0       0       0   
6       0       0       0       0       0       0       0       0       0   
7       0       0       0       0       0       0       0       0       0   
8       0       0       0       0       0       0       0       0       0   
9       0       0       0       0       0       0       0       0       0   

   pixel10  ...  pixel775  pixel776  pixel777  pixel778  pixel779  pixel780  \
0        0  ...         0         0         0         0         0        

In [7]:
print(X.shape)

(5000, 784)


In [8]:
print(X.columns)

Index(['pixel1', 'pixel2', 'pixel3', 'pixel4', 'pixel5', 'pixel6', 'pixel7',
       'pixel8', 'pixel9', 'pixel10',
       ...
       'pixel775', 'pixel776', 'pixel777', 'pixel778', 'pixel779', 'pixel780',
       'pixel781', 'pixel782', 'pixel783', 'pixel784'],
      dtype='object', length=784)


In [9]:
print(X.dtypes)

pixel1      int64
pixel2      int64
pixel3      int64
pixel4      int64
pixel5      int64
            ...  
pixel780    int64
pixel781    int64
pixel782    int64
pixel783    int64
pixel784    int64
Length: 784, dtype: object


In [10]:
print(X.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Columns: 784 entries, pixel1 to pixel784
dtypes: int64(784)
memory usage: 29.9 MB
None


<font color='red'>Quelle est la structure de la base X ? </font>

<font color='lightbluereen'>5 000 lignes X 784 colonnes</font>

<font color='red'>Que représentent les colonnes ?</font>

<font color='lightgreen'>Elles représentent des pixels (surement d'une image)</font>

<font color='red'>Que représentent les colonnes ?</font>

Dans quel format les données sont-elles stockées ?